### DISCUSSION TITLE
When do the sensors calculate?

### DISCUSSION THREAD

**Discussion by djdarlo** [BottlecapDave/HomeAssistant-OctopusEnergy#201] [Q&A]
I've setup two target sensors this morning. One 1 hour and one 4 hour. When do these calculate? I was expecting them to do so every half hour maybe but they don't seem to be populating anything in the attributes tab.

**Comment by BottlecapDave** [comment]
Hi there!

When did you setup your sensors and do you have any errors associated in the logs?

As per the examples. If you do not set a start/end time and your target rate sensors are not rolling then they will find the cheapest period between 00:00:00 and 23:59:59, which may be in the past and therefore you've missed the window for today. The sensors are calculated every minute assuming that rates are available for the next day.

**Comment by djdarlo** [comment]
That makes perfect sense - thank you. I haven’t set any start and end dates
so it will recalc tomorrow. Got it. :)

D
